src/tools/knowledge_graph_tools.py

```
@@ -0,0 +1,586 @@
+    Returns:
+        str: A descriptive message to help diagnose Neo4j connection or authentication issues.
+    """
+    error_str = str(error).lower()
+    if "authentication" in error_str or "unauthorized" in error_str:
+        return "Neo4j authentication failed. Check NEO4J_USER and NEO4J_PASSWORD."
+    elif "connection" in error_str or "refused" in error_str or "timeout" in error_str:
+        return "Cannot connect to Neo4j. Check NEO4J_URI and ensure Neo4j is running."
+    elif "database" in error_str:
+        return "Neo4j database error. Check if the database exists and is accessible."
+    else:
+        return f"Neo4j error: {str(error)}"
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "return", remove the leading "el" from "elif"

(R1705)